### PR TITLE
Migrated brave theme prefs from profile to local state

### PR DESCRIPTION
**Pull request by simonhong**
With this, we can fix brave theme related bugs with multi profile
because all profiles use browser-wide prefs.
So far, system theme could be changed whenever new profile is used
because brave theme was profile prefs but system theme is browser
wide setting.
And did cleanup and refactoring brave theme related implementations.

Refactoring point is separation of theme managing and dark mode managing.
So far, theme service has all dark mode handling logic. Mixing of dark mode and theme concepts
could cause confusion and made difficult to read the code.
I think theme service should handle browser theme based on underlying dark mode status.
So, all dark mode logic is separated from theme service in `dark_mode` namespace.

New `kBraveDarkMode` prefs are introduced and previous prefs are deprecated.
The value of `kBraveThemeType` is migrated to `kBraveDarkMode`.

fix brave/brave-browser#5373
fix brave/brave-browser#5557

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [x] Linux
  - [x] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Open browser and check current theme type is same with previously used type
2. Create new profile and check its new window also has same theme option
3. Check changing option from one of profiles affects all other profile's window.
    All should have same theme type.
 
Refer issues for bug scenarios

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
Kindly ping to owner's review @bridiver  :)

File: browser/BUILD.gn [hunk 1]
```diff
@@ -191,6 +190,7 @@ source_set("browser_process") {
       "importer",
       "infobars",
       "//brave/app:brave_generated_resources_grit",
+      "//brave/browser/themes",
```
Comment (bridiver) [inline browser/BUILD.gn]: why this change? Usually relative paths are used for subdirs

File: chromium_src/chrome/browser/prefs/browser_prefs.cc [hunk 2]
```diff
@@ -13,6 +13,10 @@
 #include "brave/browser/widevine/widevine_utils.h"
 #endif
 
+#if !defined(OS_ANDROID)
```
Comment (bridiver) [inline chromium_src/chrome/browser/prefs/browser_prefs.cc]: This file seems to be inconsistently guarded. When it's used in so many places I think it's generally better to guard it internally with no-op in android rather that try to guard it in a dozen different places

File: browser/BUILD.gn (hunk 1 above)
Comment (simonhong) [reply browser/BUILD.gn]: Fixed to `themes`.

File: chromium_src/chrome/browser/prefs/browser_prefs.cc (hunk 2 above)
Comment (simonhong) [reply chromium_src/chrome/browser/prefs/browser_prefs.cc]: Yup, made this file no-op for android.
